### ISSUE TITLE
ScriptTimeoutError: Error 1502/ EOFError: Error 2030

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1124]
@cwillisf I wasn't sure which Sentry errors would be helpful for a "Sentry-based cleanup", but I figured at least opening one case would guarantee we revisit the topic soon and it doesn't get lost, even if these weren't good errors to pick. : )

I picked two kinds of error that seemed to occur a lot (hopefully one of them will make sense for the cleanup):

**ScriptTimeoutError**
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105147916/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105148061/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105149821/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105148402/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105153206/

**EOFError**
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105150254/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105150253/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105150259/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105150190/
- https://app.getsentry.com/scratch-team/scratch-flash/issues/105186072/

**Comment by cwillisf** [comment]
I went through the ones that still exist and tried to categorize them and guess what could cause them.

`ScriptTimeoutError` (Error 1502):
- In general, the call stacks for this error could be deceptive. This error occurs when the browser decides that Flash has been taking too long. If our code is truly "frozen" then the call stack will point to an infinite loop or something similar. However, if the project is gradually slowing down as the user adds more and more stuff to the project, then certain "heavy" operations (like a UI refresh, etc.) may just barely cross the browser's threshold. In this case, the browser might interrupt Flash in just about any part of a regular frame step, leading to a call stack that isn't very useful. For the items below I've assumed that this isn't the case, instead assuming that the call stack indicates the actual location of the problem.
- An HTTP extension took too long to process something. There are no official HTTP extensions, so this is an issue for the extension author. Unfortunately it's not clear which extension was running at the time.
- The Flash sound system took too long to accept sound samples. We don't have any variable-length loops in this area of the code, so the only thing I can come up with is that it may be possible for Flash's or even the operating system's sound subsystem to freeze. Note that this error has happened on both Windows and OS X computers.
- The sprite library took too long to refresh. It's not clear to me how this code could take a long time.
- Calculating the bounds of a sprite took too long. For a very large sprite on a slow computer, it may not be possible to prevent this problem. That said, we could try to make sure not to calculate the bounds for multiple sprites in the same frame and maybe a few other things to reduce the chances of this happening.

`EOFError` (Error 2030):
- Several of these errors have disappeared from Sentry. I guess we fixed them? :bowtie:
- The `play sound` or `play sound and wait` tried to play a sound that has truncated (or possibly empty) WAV file contents. I recently made a change (with #1225) which should prevent this from happening, but I see that the error has happened even after that change. I would love to find a way to reproduce this issue.

**Comment by cwillisf** [comment]
I talked this over with @thisandagain and we concluded that most of the `ScriptTimeoutError` instances are not worth pursuing at the moment. The exception is that if we see `ScriptTimeoutError` instances that look like they're due to bounds calculations for multiple sprites in the same frame; I looked for that and couldn't find any such cases in Sentry.
Since #1228 has already been split off into a separate issue, I'm closing this one.
It looks like I should try to fix #1223, though, which is another case of the same sort of thing.
